### PR TITLE
Updates to ExternalNetwork strategy

### PR DESCRIPTION
**Pull request by steve-gregory**
We made some incorrect assumptions about the external network strategy.

After some amount of clarification, these are the changes that were required in order to get ExternalNetwork support setup.

### CODE REVIEW

File: service/networking.py [hunk 1]
```diff
@@ -229,19 +236,10 @@ def __init__(self, identity):
...
-
-    def get_or_create_network(self, network_driver, user_neutron, network_name):
-        # Step 1. Does public network exist?
+    def get_public_network(self, network_driver):
+        """
+        This method is special to ExternalNetwork
+        """
         public_network = network_driver.find_network(
             self.external_network_name)
         if type(public_network) == list:
```
Comment (cdosborn) [inline service/networking.py]: What if public_network is an empty list?

Comment (steve-gregory) [inline service/networking.py]: validate() call should raise exception before you get here

**Comment by cdosborn** [comment]
ill tack on a commit here, its easier sometimes to show code.

**Comment by cdosborn** [comment]
I did notice that the `gateway` variable was not being used.

**Comment by steve-gregory** [comment]
@cdosborn Thanks! This is where I was intending to go after I had commited the "parity" version, so we're both on the same page

**Comment by steve-gregory** [comment]
TODO:

- [ ] Test this version of moc-master and be sure it doesn't break networking

**Comment by cdosborn** [comment]
i'll buy you mcdees if my code works.

**Comment by cdosborn** [comment]
@o@ IT WORKED???

**Pull request merged**